### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by tomerXarchimedes**
Added liquidity to 3CRV/lvUSD pool on a mainnet fork :-)

Online review: Jason

### CODE REVIEW

File: integrationTests/i3CRVlvUSDCurvePool.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
+// We import Chai to use its asserting functions here.
+import { expect } from "chai";
+import { Contract, Signer } from "ethers";
+import { ethers } from "hardhat";
+import { abilvUSD, abilvUSD3CRVPool, abiUSDC, abiZap } from "../test/ABIs";
```
Comment (yotamdaniel) [inline integrationTests/i3CRVlvUSDCurvePool.ts:5]: What is zap?

File: integrationTests/i3CRVlvUSDCurvePool.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import { addressZap, address3CRV, addressUSDC, defaultBlockNumber, helperResetNetwork, address3CRVlvUSDPool } from "../test/MainnetHelper";
+import { impersonateAccount, fundAccount, stopImpersonate, addresslvUSDToken } from "./IntegrationTestContext";
+
+const About4700 = "0x1000000000000000000";
+
+const addr1 = "0x55fe002aeff02f77364de339a1292923a15844b8"; // Circles address
```
Comment (yotamdaniel) [inline integrationTests/i3CRVlvUSDCurvePool.ts:11]: What is circles?

File: integrationTests/i3CRVlvUSDCurvePool.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+        // grab the "before" balances so we can check they increase after adding liquidity
+        const balancelvUSD = await contractlvUSD3CRVPool.balances(0);
+        const balanceUSDC = await contractlvUSD3CRVPool.balances(1);
+
+        // Seed 3CRV/lvUSD pool via Zap
+        // Indexes: 0 = lvusd, 1 = dai, 2 = usdc, 3 = usdt
+        // Indexes 123 are the 3curve token
+        const zap = await ethers.getContractAt(abiZap, addressZap);
+        const coins = [ethers.utils.parseUnits("100", 18), "0x0", ethers.utils.parseUnits("100", 6), "0x0"];
```
Comment (yotamdaniel) [inline integrationTests/i3CRVlvUSDCurvePool.ts:76]: What is "0x0" used for?

File: integrationTests/i3CRVlvUSDCurvePool.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+        // approve Zap contract to grab lvUSD and USDC from addr1
+        await contractlvUSDToken.connect(signerAddr1).approve(addressZap, tenK18Decimal);
+        await contractUSDC.connect(signerAddr1).approve(addressZap, tenK6Decimal);
+
+        // get 3CRV/lvUSD contract
+        contractlvUSD3CRVPool = await ethers.getContractAt(abilvUSD3CRVPool, address3CRVlvUSDPool);
+
+        // grab the "before" balances so we can check they increase after adding liquidity
+        const balancelvUSD = await contractlvUSD3CRVPool.balances(0);
```
Comment (yotamdaniel) [inline integrationTests/i3CRVlvUSDCurvePool.ts:69]: Change name to balancePoolLvUSD and same for balanceUSDC?

File: integrationTests/i3CRVlvUSDCurvePool.ts [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        const balanceUSDC = await contractlvUSD3CRVPool.balances(1);
+
+        // Seed 3CRV/lvUSD pool via Zap
+        // Indexes: 0 = lvusd, 1 = dai, 2 = usdc, 3 = usdt
+        // Indexes 123 are the 3curve token
+        const zap = await ethers.getContractAt(abiZap, addressZap);
+        const coins = [ethers.utils.parseUnits("100", 18), "0x0", ethers.utils.parseUnits("100", 6), "0x0"];
+        await zap.connect(signerAddr1).add_liquidity(address3CRVlvUSDPool, coins, 0);
+
+        expect(await contractlvUSD3CRVPool.balances(0)).to.be.gt(balancelvUSD);
```
Comment (yotamdaniel) [inline integrationTests/i3CRVlvUSDCurvePool.ts:79]: we can add an assert that check that what was add was almost as close to 1:1. Use `expect(value).to.be.closeTo(BigNumber.from(...), margin)`

File: integrationTests/i3CRVlvUSDCurvePool.ts (hunk 2 above)
Comment (tomerXarchimedes) [reply integrationTests/i3CRVlvUSDCurvePool.ts:11]: Forgot the ' - it's Circle's (Coinbase and Circle are the main companies behind USDC)

File: integrationTests/i3CRVlvUSDCurvePool.ts (hunk 4 above)
Comment (tomerXarchimedes) [reply integrationTests/i3CRVlvUSDCurvePool.ts:69]: ack

File: integrationTests/i3CRVlvUSDCurvePool.ts (hunk 3 above)
Comment (tomerXarchimedes) [reply integrationTests/i3CRVlvUSDCurvePool.ts:76]: The amount of tokens - I'll improve the comment

**Pull request merged**